tuf/ngclient/_internal/trusted_metadata_set.py

```diff
@@ -196,7 +196,7 @@ def update_root(self, data: bytes) -> None:
         logger.debug("Updating root")
 
         try:
-            new_root = Metadata[Root].from_bytes(data)
+            new_root: Metadata[Root] = Metadata[Root].from_bytes(data)
```

Comment:
These type hints for the IDE are starting to look quite cumbersome. Can the IDE not recognise the type hint on the right-hand side of the assignment operator? Do we really need them?😕

Reply:
Now that you made me have another look :roll_eyes: .... it turned out I wasn't able to annotate the return type of the `from_*` methods with the generic type 'T' because `from_bytes` was a static method. So I amended the last commit and I think this one now looks better and does the job: https://github.com/theupdateframework/tuf/pull/1489/commits/ba6c672506ec386c1ccccb195a33e236f14b4ee7